### PR TITLE
feat: Adding Windows Server version 1903 support

### LINKED ISSUE #1462: Add latest Windows Server SAC (1903) support
**Describe the request**
Windows Server version 1903 is now released and supported, and Windows Server version 1709 support was dropped. Update AKS-Engine to match

https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info

**Explain why AKS Engine needs it**

People want to try Windows Server version 1903

### PR DESCRIPTION
**Pull request by PatrickLang**
**Reason for Change**:

This adds support for Windows Server version 1903, along with a sample and brief update to clusterdefinition.md. I also cleaned up the container version references and added comments to help identify what containers to use with the next Windows OS version change.

**Issue Fixed**:

resolves #1462

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by PatrickLang** [comment]
/hold

I'm manually running `make test-kubernetes` on that sample included to make sure E2E tests pass with Windows Server version 1903 + Kubernetes 1.15.0-beta.2

**Comment by PatrickLang** [comment]
cc @kkmsft

File: docs/topics/clusterdefinitions.md [hunk 1]
```diff
@@ -645,7 +645,7 @@ https://{keyvaultname}.vault.azure.net:443/secrets/{secretName}/{version}
 | windowsSku                       | no       | SKU usedto find Windows VM to deploy from marketplace. Default: `Datacenter-Core-1809-with-Containers-smalldisk` |
 | imageVersion                     | no       | Specific image version to deploy from marketplace.  Default: `latest` |
 | windowsImageSourceURL            | no       | Path to an existing Azure storage blob with a sysprepped VHD. This is used to test pre-release or customized VHD files that you have uploaded to Azure. If provided, the above 4 parameters are ignored. |
-| sshEnabled                       | no       | If set to `true`, OpenSSH will be installed on windows nodes to allow for ssh remoting. **Only for Windows version 1809 or 2019** . The same SSH authorized public key(s) will be added from [linuxProfile.ssh.publicKeys](#linuxProfile) |
+| sshEnabled                       | no       | If set to `true`, OpenSSH will be installed on windows nodes to allow for ssh remoting. **Only for Windows version 1809/2019 or later** . The same SSH authorized public key(s) will be added from [linuxProfile.ssh.publicKeys](#linuxProfile) |
```
Comment (kkmsft) [inline docs/topics/clusterdefinitions.md]: Just for understanding - is this because ssh by default is supported in 1809/2019 or later ?

Comment (PatrickLang) [reply docs/topics/clusterdefinitions.md]: yes, it's available but not started by default. Setting this option will finish setting up SSHD and start the service

File: test/e2e/kubernetes/job/job.go [hunk 2]
```diff
@@ -57,6 +60,45 @@ func CreateJobFromFile(filename, name, namespace string) (*Job, error) {
 	return job, nil
 }
 
+// CreateWindowsJobFromTemplate will create a Job from file with a name
+func CreateWindowsJobFromTemplate(filename, name, namespace, servercore string) (*Job, error) {
```
Comment (PatrickLang) [inline test/e2e/kubernetes/job/job.go]: @jackfrancis - looks like you were the last to update the `validate-dns-windows.yaml` area. What do you think of this approach?

File: parts/k8s/windowskubeletfunc.ps1 [hunk 3]
```diff
@@ -130,30 +130,39 @@ users:
...
-        "mcr.microsoft.com/nanoserver-insider"
-    }
 
-    "FROM $($windowsBase)" | Out-File -encoding ascii -FilePath Dockerfile
-    "CMD cmd /c ping -t localhost" | Out-File -encoding ascii -FilePath Dockerfile -Append
-    docker build -t kubletwin/pause .
+    # Reference for these tags: curl -L https://mcr.microsoft.com/v2/k8s/core/pause/tags/list
+    # Then docker run --rm mplatform/manifest-tool inspect mcr.microsoft.com/k8s/core/pause:<tag>
+
+    $windowsBase = switch ($computerInfo.WindowsVersion) {
```
Comment (PatrickLang) [inline parts/k8s/windowskubeletfunc.ps1]: I'm debating whether or not this version-specific behavior should be in scripts on the node, or part of AKS-Engine.

Pros for AKS-Engine:

- can override pause image with another one in ACR, Docker Hub, or elsewhere

Pros for this script:

- AKS-Engine could deploy multiple Windows agent pools and each node will automatically figure out the right image

**Comment by PatrickLang** [comment]
I'm running a second test with the new apimodel attached in the PR. I ran it against an earlier commit and it passed but double checking now.

**Comment by PatrickLang** [comment]
AKS-Engine E2E passes with the new definition checked in. 🛳 it

```
Ran 29 of 45 Specs in 1565.664 seconds
SUCCESS! -- 29 Passed | 0 Failed | 0 Pending | 16 Skipped
PASS  
```

**Pull request merged**